net: ethernet: Allow to disable the auto-start of ethernet interfaces

jetstreeam opened the pull request:
Adds CONFIG_ETH_NET_IF_NO_AUTO_START to allow pre-configuration of Ethernet interfaces (e.g., filters, EUI-64) before they become operational. When enabled, net_if_up() must be explicitly called by the application.

maass-hamburg commented on drivers/ethernet/Kconfig:
```diff
@@ -99,3 +99,12 @@ config ETH_INIT_PRIORITY
+
+config ETH_NET_IF_NO_AUTO_START
+	bool "Disable Ethernet interface auto-start"
+	help
```
Should add a
```
depends on NET_L2_ETHERNET
```
or at least a
```
depends on NETWORKING
```
Don't want to "polute" the Kconfig, when Ethernet is not activated.

jukkar commented on drivers/ethernet/Kconfig:
```diff
@@ -99,3 +99,12 @@ config ETH_INIT_PRIORITY
+	help
+	  This option allows user to set any configuration before the interface
+	  becomes operational. For instance, the EUI-64 value can be configured using
+	  net_if_set_link_addr(iface, mac, 8, NET_LINK_ETHERNET).
```
Isn't there a typo in the help text, the Ethernet link addresses are 6 bytes long.

jetstreeam replied:
thanks, updated.

jetstreeam committed:
net: ethernet: Allow to disable the auto-start of ethernet interfaces

Adds CONFIG_ETH_NET_IF_NO_AUTO_START to allow
pre-configuration of Ethernet interfaces (e.g., filters, mac)
before they become operational. When enabled, net_if_up() must be
explicitly called by the application.

Signed-off-by: Jeremy Truttmann <jetstreeam.code@outlook.com>
